rewriter/initializer.cc

```diff
@@ -0,0 +1,128 @@
+#include "rewriter/initializer.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/core.h"
+
+#include <utility>
+
+using namespace std;
+namespace sorbet::rewriter {
+
+// this only looks for sigs of the 'correct' shape (i.e. sigs with blocks that contain a send that also end with `void`)
+bool isSig(const ast::Send *send) {
+    if (send->fun != core::Names::sig()) {
+        return false;
+    }
+    if (send->block.get() == nullptr) {
+        return false;
+    }
+    auto nargs = send->args.size();
+    if (nargs != 0 && nargs != 1) {
+        return false;
+    }
+    auto block = ast::cast_tree<ast::Block>(send->block.get());
+    ENFORCE(block);
+    auto body = ast::cast_tree<ast::Send>(block->body.get());
+    if (!body) {
+        return false;
+    }
+    if (body->fun != core::Names::void_()) {
+        return false;
+    }
+
+    return true;
+}
+
+bool isCopyableType(const ast::Expression *typeExpr) {
+    auto send = ast::cast_tree_const<ast::Send>(typeExpr);
+    if (send && send->fun == core::Names::typeParameter()) {
+        return false;
+    }
+    return true;
+}
+
+// if expr is of the form `@var = local`, and `local` is typed, then replace it with with `@var = T.let(local,
+// type_of_local)`
+void maybeAddLet(core::MutableContext ctx, ast::Expression *expr,
+                 const UnorderedMap<core::NameRef, ast::Expression *> &argTypeMap) {
+    auto assn = ast::cast_tree<ast::Assign>(expr);
+    if (!assn) {
```

Comment:
This is non-blocking feedback, but I vastly prefer the style `assn == nullptr` to `!assn`.

Reply:
I use both inconsistently in this file, so I'll change them all to `== nullptr`.